bindings/matrix-sdk-ffi/src/timeline/mod.rs

```diff
@@ -1021,100 +1029,69 @@ impl From<SdkShieldState> for ShieldState {
-    }
-
-    pub fn timestamp(&self) -> u64 {
-        self.0.timestamp().0.into()
-    }
+#[derive(Clone, uniffi::Record)]
+pub struct EventTimelineItem {
+    is_local: bool,
+    is_remote: bool,
+    transaction_id: Option<String>,
+    event_id: Option<String>,
+    sender: String,
+    sender_profile: ProfileDetails,
+    is_own: bool,
+    is_editable: bool,
+    content: TimelineItemContent,
+    timestamp: u64,
+    reactions: Vec<Reaction>,
+    debug_info_provider: Arc<EventTimelineItemDebugInfoProvider>,
+    local_send_state: Option<EventSendState>,
+    read_receipts: HashMap<String, Receipt>,
+    origin: Option<EventItemOrigin>,
+    can_be_replied_to: bool,
+    message_shield: Option<ShieldState>,
+}
 
-    pub fn reactions(&self) -> Vec<Reaction> {
-        self.0
+impl From<matrix_sdk_ui::timeline::EventTimelineItem> for EventTimelineItem {
+    fn from(value: matrix_sdk_ui::timeline::EventTimelineItem) -> Self {
+        let reactions = value
             .reactions()
             .iter()
             .map(|(k, v)| Reaction {
                 key: k.to_owned(),
                 senders: v
-                    .iter()
+                    .into_iter()
                     .map(|(sender_id, info)| ReactionSenderData {
                         sender_id: sender_id.to_string(),
                         timestamp: info.timestamp.0.into(),
                     })
                     .collect(),
             })
-            .collect()
-    }
-
-    pub fn debug_info(&self) -> EventTimelineItemDebugInfo {
-        EventTimelineItemDebugInfo {
-            model: format!("{:#?}", self.0),
-            original_json: self.0.original_json().map(|raw| raw.json().get().to_owned()),
-            latest_edit_json: self.0.latest_edit_json().map(|raw| raw.json().get().to_owned()),
+            .collect();
+        let debug_info_provider = Arc::new(EventTimelineItemDebugInfoProvider(value.clone()));
+        let read_receipts =
+            value.read_receipts().iter().map(|(k, v)| (k.to_string(), v.clone().into())).collect();
+        Self {
+            is_local: value.is_local_echo(),
+            is_remote: !value.is_local_echo(),
+            transaction_id: value.transaction_id().map(|t| t.to_string()),
+            event_id: value.event_id().map(|e| e.to_string()),
+            sender: value.sender().to_string(),
+            sender_profile: value.sender_profile().into(),
+            is_own: value.is_own(),
+            is_editable: value.is_editable(),
+            content: value.content().clone().into(),
+            timestamp: value.timestamp().0.into(),
+            reactions,
+            debug_info_provider,
+            local_send_state: value.send_state().map(|s| s.into()),
+            read_receipts,
+            origin: value.origin(),
+            can_be_replied_to: value.can_be_replied_to(),
+            message_shield: value.get_shield(false).map(Into::into),
```

Comment:
Maybe I can create another provider here, as I did for the debug info. I wonder if it's better to wrap the ui `EventTimelineItem` into an `Arc` instead of cloning them so we can have it in both structs.